src/Ninject.Web.AspNetCore/Planning/ParameterTargetWithKeyedSupport.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.DependencyInjection;
+using Ninject.Activation;
+using Ninject.Parameters;
+using Ninject.Planning.Bindings;
+using Ninject.Planning.Targets;
+using Ninject.Web.AspNetCore.Parameters;
+using Ninject.Web.AspNetCore.RequestActivation;
+
+namespace Ninject.Web.AspNetCore.Planning
+{
+	public class ParameterTargetWithKeyedSupport : ParameterTarget, ITarget
+	{
+		public ParameterTargetWithKeyedSupport(MethodBase method, ParameterInfo site) : base(method, site)
+		{
+		}
+
+		public override bool HasDefaultValue
+		{
+			get
+			{
+				var result = this.Site.HasDefaultValue;
+#if NET8_0_OR_GREATER
+				// ensure that constructor scorer knows that we have a default value for parameters decorated with ServiceKey
+				// as the DefaultValueBindingResolver is only a MissingBindingResolver, the
+				// ParameterTargetWithKeyedSupport.ResolveWithin method already
+				// provided a default value before any Ninject resolution for the value happens.
+				result = result || GetCustomAttributes(typeof (ServiceKeyAttribute), true)?.Length > 0;
+#endif
+				return result;
+			}
+		}
+
+		/// <summary>
+		/// MethodInjectionStrategy.GetMethodArguments calls ITarget.ResolveWithin.
+		/// As we can't override the base implementation as it is not virtual, the
+		/// explicit interface implementation helps to still delegate the resolution to here.
+		/// </summary>
+		object ITarget.ResolveWithin(IContext parent)
+		{
+#if NET8_0_OR_GREATER
+			var serviceKeyAttributes = GetCustomAttributes(typeof (ServiceKeyAttribute), true) as ServiceKeyAttribute[];
+			if (serviceKeyAttributes?.Length > 0)
+			{
+				return ResolveServiceKeyValue(parent);
+			}
+
+			var keyedattributes = GetCustomAttributes(typeof (FromKeyedServicesAttribute), true) as FromKeyedServicesAttribute[];
+			if (keyedattributes?.Length > 0)
+			{
+				return ResolveFromKeyedService(parent, keyedattributes[0]);
+			}
+#endif
+			return base.ResolveWithin(parent);
+		}
+
+#if NET8_0_OR_GREATER
+		private object ResolveFromKeyedService(IContext parent, FromKeyedServicesAttribute keyedattribute)
+		{
+			var fromKeyedServiceValue = DeterimeFromKeyedServiceValue(keyedattribute, parent.Parameters);
```

Comment:
Have to adapt the logic form base.ResolveWithin(parent) to correctly handle fromKeyedService attributes and default values. The implementation itself is quite similar still to the base implementation.